src/main/java/com/radixdlt/crypto/hdwallet/BitcoinJBIP32Path.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.crypto.hdwallet;
+
+import com.google.common.base.Objects;
+import org.bitcoinj.crypto.ChildNumber;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A BIP32 path wrapping underlying implementation using BitcoinJ.
+ */
+final class BitcoinJBIP32Path implements HDPath {
+
+	private static final String BIP32_HARDENED_MARKER_BITCOINJ = "H";
+
+	private final org.bitcoinj.crypto.HDPath path;
+
+	private BitcoinJBIP32Path(org.bitcoinj.crypto.HDPath path) {
+		this.path = path;
+	}
+
+	static BitcoinJBIP32Path fromPath(HDPath path) {
+		try {
+			return fromString(path.toString());
+		} catch (HDPathException e) {
+			throw new IllegalStateException("String representation of any path should be correct.", e);
+		}
+	}
+
+	static BitcoinJBIP32Path fromString(String path) throws HDPathException {
+		HDPaths.validateHDPathString(path);
+		return new BitcoinJBIP32Path(org.bitcoinj.crypto.HDPath.parsePath(toBitcoinJPath(path)));
+	}
+
+	private static String toBitcoinJPath(String standardPath) {
+		// For some reason BitcoinJ chose to not use standard notation of hardened path components....
+		return standardPath.replace(HDPaths.BIP32_HARDENED_MARKER_STANDARD, BIP32_HARDENED_MARKER_BITCOINJ);
+	}
+
+	private static String standardizePath(String nonStandardPath) {
+		// For some reason BitcoinJ chose to not use standard notation of hardened path components....
+		return nonStandardPath.replace(BIP32_HARDENED_MARKER_BITCOINJ, HDPaths.BIP32_HARDENED_MARKER_STANDARD);
+	}
+
+	private int indexOfLastComponent() {
+		if (depth() == 0) {
+			throw new IllegalStateException("Trying to access component of a BIP32 path with 0 depth, this is undefined.");
```

Comment:
Only slightly unsure about this.

In short: All relevant HDPaths have at least a depth 1. However, the empty path "" or "/" results in a special `HDPath` having depth 0. This is a valid path - theoretically, but never really used. 

So throwing an `IllegalStateException` in the case anyone tries to access properties which does not really exist for these theoretical empty paths....